src/Microsoft.ML.TorchSharp/Roberta/QATrainer.cs

```diff
@@ -0,0 +1,938 @@
+
+            /// <summary>
+            /// How often to log the loss.
+            /// </summary>
+            public int LogEveryNStep = 50;
+
+            public Options()
+            {
+                EncoderOutputDim = 768;
+                PoolerDropout = 0;
+                ModelType = BertModelType.Roberta;
+                TaskType = BertTaskType.QuestionAnswering;
+            }
+        }
+
+        internal static class DefaultColumnNames
+        {
+            public const string Context = "Context";
+            public const string Question = "Question";
+            public const string Answer = "Answer";
+            public const string TrainingAnswer = "TrainingAnswer";
+            public const string AnswerIndex = "AnswerStart";
+            public const int TopKAnswers = 3;
+            public const string Score = "Score";
+        }
+
+        private protected readonly IHost Host;
+        internal readonly Options Option;
+        private const string ModelUrl = "models/pretrained_Roberta_encoder.tsm";
+
+        internal QATrainer(IHostEnvironment env, Options options)
+        {
+            Host = Contracts.CheckRef(env, nameof(env)).Register(nameof(QATrainer));
+            Contracts.Assert(options.MaxEpoch > 0);
+            Contracts.AssertValue(options.ContextColumnName);
+            Contracts.AssertValue(options.QuestionColumnName);
+            Contracts.AssertValue(options.TrainingAnswerColumnName);
+            Contracts.AssertValue(options.AnswerIndexStartColumnName);
+            Contracts.AssertValue(options.ScoreColumnName);
+            Contracts.AssertValue(options.PredictedAnswerColumnName);
+
+            Option = options;
+        }
+
+        internal QATrainer(IHostEnvironment env,
+            string contextColumnName = DefaultColumnNames.Context,
+            string questionColumnName = DefaultColumnNames.Question,
+            string trainingAnswerColumnName = DefaultColumnNames.TrainingAnswer,
+            string answerIndexColumnName = DefaultColumnNames.AnswerIndex,
+            string predictedAnswerColumnName = DefaultColumnNames.Answer,
+            string scoreColumnName = DefaultColumnNames.Score,
+            int topk = 3,
+            int batchSize = 4,
+            int maxEpochs = 10,
+            IDataView validationSet = null,
+            BertArchitecture architecture = BertArchitecture.Roberta) :
+            this(env, new Options
+            {
+                ContextColumnName = contextColumnName,
+                QuestionColumnName = questionColumnName,
+                TrainingAnswerColumnName = trainingAnswerColumnName,
+                AnswerIndexStartColumnName = answerIndexColumnName,
+                PredictedAnswerColumnName = predictedAnswerColumnName,
+                ScoreColumnName = scoreColumnName,
+                TopKAnswers = topk,
+                BatchSize = batchSize,
+                MaxEpoch = maxEpochs,
+                ValidationSet = validationSet
```

Comment:
the same question as https://github.com/dotnet/machinelearning/pull/6760/files#r1259001582